### PR TITLE
refactor: extract init from node-core

### LINKED ISSUE #8171: extract node-core/init.rs from node-core
### Describe the feature

these operations are out of place here:

https://github.com/paradigmxyz/reth/blob/aa07257d3ba071188b7b7e372856b50be094e8ba/crates/node-core/src/init.rs

they also can't be included in any existing storage crates, like db/provider and should be moved to `reth-db-{common,core,init,ops}`. Any other naming ideas @Rjected ?
leaning towards reth-db-common

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by wlawt**
Resolves #8171 .

### Overview

Introduces the `reth-db-common` crate and moves `node-core/init.rs` into this new crate.

### Test Plan

- Ran unit tests

### CODE REVIEW

**Comment by mattsse** [comment]
ah cool, I was looking for this actually.
I vaguely remember that there was a previous attempt, could be wrong

File: crates/db-common/src/lib.rs
Comment (mattsse) [inline crates/db-common/src/lib.rs:1]: this needs the regular lib.rs header including the cfg for unused crates

**Review by mattsse — CHANGES_REQUESTED** [review]
nice!

only have a style nit

File: crates/db-common/src/lib.rs
Comment (wlawt) [reply crates/db-common/src/lib.rs:1]: done! also cleaned up the unused dependencies

**Comment by wlawt** [comment]
how can i fix the fmt issue? when i run the `cargo fmt` command used in the CI test, it changes a lot of unrelated files cc @mattsse , thank u!

File: Cargo.toml [hunk 1]
```diff
@@ -95,6 +95,7 @@ members = [
     "examples/txpool-tracing/",
     "testing/ef-tests/",
     "testing/testing-utils",
+    "crates/db-common",
```
Comment (mattsse) [inline Cargo.toml:98]: this should be moved into the crates/storage folder

also please put the new entry in the alphabetically correct position.

File: Cargo.toml [hunk 2]
```diff
@@ -284,6 +285,7 @@ reth-tracing = { path = "crates/tracing" }
 reth-transaction-pool = { path = "crates/transaction-pool" }
 reth-trie = { path = "crates/trie" }
 reth-trie-parallel = { path = "crates/trie-parallel" }
+reth-db-common = { path = "crates/db-common" }
```
Comment (mattsse) [inline Cargo.toml:288]: alphabetical order

File: crates/node-core/Cargo.toml [hunk 3]
```diff
@@ -35,11 +35,8 @@ reth-network-api.workspace = true
 reth-evm.workspace = true
 reth-engine-primitives.workspace = true
 reth-tasks.workspace = true
-reth-trie.workspace = true
 reth-consensus-common.workspace = true
 reth-beacon-consensus.workspace = true
-reth-etl.workspace = true
-reth-codecs.workspace = true
```
Comment (mattsse) [inline crates/node-core/Cargo.toml:42]: great!

**Review by mattsse — CHANGES_REQUESTED** [review]
last pedantic nit

to format please run `cargo +nightly fmt --all`

**Review by mattsse — APPROVED** [review]
nice

**Pull request merged**